desktop/src/main/java/bisq/desktop/primary/main/content/social/chat/ChatController.java

```diff
@@ -273,10 +273,19 @@ public void onReply(ChatMessage chatMessage) {
         }
     }
 
-    public void onSendPrivateMessage(ChatMessage chatMessage) {
-        if (!model.isMyMessage(chatMessage)) {
-            // todo
+    /**
+     * open a private channel to specified user. Automatically select it so user is ready to type the message to send.
+     * @param chatMessage
+     */
+    public void addPrivateChannel(ChatMessage chatMessage) {
+        if (model.isMyMessage(chatMessage)) {
+            return; // should never happen as the button for opening the channel should not appear
+            // but kept here for double safety
         }
+        ChatUser peerUser = chatMessage.getChatUser();
+        String channelId = PrivateChannel.createChannelId(peerUser, userProfileService.getPersistableStore().getSelectedUserProfile().get());
+        PrivateChannel channel = chatService.getOrCreatePrivateChannel(channelId, peerUser);
```

Comment:
I think the PrivateChannel.createChannelId part could be moved to the domain to be handled inside getOrCreatePrivateChannel.